src/routes/(console)/project-[region]-[project]/sites/create-site/deploy/+page.svelte

```diff
@@ -0,0 +1,365 @@
+    import {
+        Fieldset,
+        Layout,
+        Icon,
+        Typography,
+        Input,
+        Tag,
+        Selector
+    } from '@appwrite.io/pink-svelte';
+    import { IconGithub, IconExternalLink, IconPencil } from '@appwrite.io/pink-icons-svelte';
+    import { writable } from 'svelte/store';
+    import Domain from '../domain.svelte';
+    import { Adapter, BuildRuntime, Framework, ID } from '@appwrite.io/console';
+    import { CustomId } from '$lib/components';
+    import { getFrameworkIcon } from '$lib/stores/sites';
+    import { regionalConsoleVariables } from '$routes/(console)/project-[region]-[project]/store';
+    import { iconPath } from '$lib/stores/app';
+    import type { PageData } from './$types';
+
+    export let data: PageData;
+
+    let showExitModal = false;
+    let showCustomId = false;
+    let formComponent: Form;
+    let isSubmitting = writable(false);
+
+    // Default values
+    let name = data.repository.name;
+    let id = ID.unique();
+    let domain = data.repository.name.toLowerCase().replace(/[^a-z0-9]/g, '-');
+    let domainIsValid = true;
+    // Get URL params
+    const urlParams = new URLSearchParams($pageStore.url.search);
+    const preset = urlParams.get('preset') || 'nextjs';
+
+    // Map preset string to Framework enum
+    const presetMap = {
+        nextjs: Framework.Nextjs,
+        react: Framework.React,
+        vue: Framework.Vue,
+        nuxt: Framework.Nuxt,
+        sveltekit: Framework.Sveltekit,
+        astro: Framework.Astro,
+        vite: Framework.Vite,
+        other: Framework.Other
+    };
+    let framework = presetMap[preset.toLowerCase()] || Framework.Nextjs;
+    let branch = '';
+    let rootDir = '';
+    let variables: Array<{ key: string; value: string; secret: boolean }> = [];
+
+    // Build configuration - use from URL params or defaults
+    let installCommand = urlParams.get('install') || '';
+    let buildCommand = urlParams.get('build') || '';
+    let outputDirectory = urlParams.get('output') || '';
+
+    // Initialize environment variables from query params
+    if (data.envKeys.length > 0) {
+        variables = data.envKeys.map((key) => ({ key, value: '', secret: false }));
+    }
+
+    // Framework options
+    const frameworkOptions = [
+        { key: Framework.Nextjs, name: 'Next.js', buildRuntime: BuildRuntime.Node210 },
+        { key: Framework.React, name: 'React', buildRuntime: BuildRuntime.Node210 },
+        { key: Framework.Vue, name: 'Vue', buildRuntime: BuildRuntime.Node210 },
+        { key: Framework.Nuxt, name: 'Nuxt', buildRuntime: BuildRuntime.Node210 },
+        { key: Framework.Sveltekit, name: 'SvelteKit', buildRuntime: BuildRuntime.Node210 },
+        { key: Framework.Astro, name: 'Astro', buildRuntime: BuildRuntime.Node210 },
+        { key: Framework.Vite, name: 'Vite', buildRuntime: BuildRuntime.Node210 },
+        { key: Framework.Other, name: 'Other', buildRuntime: BuildRuntime.Static1 }
+    ];
```

Comment:
see if we can use the `Object.values` and cleanup a bit because all are Node210 except one and 2 name changes.